stdlib/json.gr

```diff
@@ -2069,3 +2069,298 @@ provide let parse: (str: String) => Result<Json, JsonParseError> = (str: String)
     }
   }
 }
+
+/**
+ * Utilities for accessing and updating JSON data.
+ *
+ * @since v0.7.0
+ */
+provide module Lenses {
+  /**
+   * A structure which provides functionality for accessing and setting JSON
+   * data.
+   *
+   * @since v0.7.0
+   */
+  provide record Lens<a, b> {
```

Comment:
Should this be `abstract` I didnt see any tests for `Lens.set` or `Lens.get` off a premade Lens.

Reply:
I think it shouldn't be abstract so that users can write their own lenses. Alternatively we could just provide a `lens` function with getter and setter arguments and make this abstract

Reply:
Also not sure what the point of testing custom lenses is, the tests are effectively testing that the lenses provided by the library work